go-controller/pkg/networkmanager/nad_controller.go

```diff
@@ -331,7 +411,43 @@ func (c *nadController) syncNAD(key string, nad *nettypes.NetworkAttachmentDefin
 		if len(oldNetwork.GetNADs()) == 0 {
 			c.networkController.DeleteNetwork(oldNetworkName)
 		} else {
-			c.networkController.EnsureNetwork(oldNetwork)
```

Comment:
I don't even understand anymore why we want to ensure old network if it has more references? the comment says
```
// remove the NAD reference from the old network and delete the network if
// it is no longer referenced
``` and that is kinda what I expected, maybe you can explain?